client/src/main.rs

```diff
@@ -64,12 +66,38 @@ async fn main_inner() -> Result<(), Error> {
         tokio::spawn(async move { rpc_server.run().await });
     }
 
+    // Vector for task monitors (Tokio task metrics)
+    let mut nimiq_task_metric = vec![];
+
     // Start consensus.
     let consensus = client.consensus().unwrap();
 
     log::info!("Spawning consensus");
-    tokio::spawn(consensus);
+    let con_metrics_monitor = tokio_metrics::TaskMonitor::new();
+    let instr_con = con_metrics_monitor.instrument(consensus);
+    tokio::spawn(instr_con);
     let consensus = client.consensus_proxy();
+    nimiq_task_metric.push(NimiqTaskMonitor {
+        name: "consensus".to_string(),
+        monitor: con_metrics_monitor,
+    });
+
```

Comment:
Having all this rather specific logic in the client/main.rs feels rather unfortunate to me, since the code here is otherwise fairly high-level. However, I also can't think of a nice solution right now. Maybe only do the instrumenting if the `metrics-server` feature is enabled, and maybe move construction of the `nimiq_task_metric` down to the metrics-server code.